### ISSUE TITLE
Spacing and style in search bar

### ISSUE THREAD

**Issue by fcoveram** [WordPress/wporg-developer#312]
As described [here](https://github.com/WordPress/wporg-developer/pull/305#issuecomment-1784796554), the search bar has new changes:

## Style
* In Article post-type, the search bar always have a border bottom of 1px in Light Grey
* When search bar is below the readcrumb, padding-top is 10px. When is not, padding-top is 20px.

![New design of search bar area in Article](https://github.com/WordPress/wporg-developer/assets/895819/9d9ad89e-b36b-4ec3-8769-c088ef97f51a)

---

## Design
[Developer mockups](https://www.figma.com/file/2WxlJFzMJvqPfZL1EkAOVp/Developer-Resources?type=design&node-id=2934-6590&mode=design&t=7w1GtutWNO2sHH72-4). Updated with these changes.

**Comment by adamwoodnz** [comment]
@fcoveram so is the search bar sticky, and is it only on this page?

Currently all parts of the header are sticky, so this addition will result in a large chunk of the page being taken up. Have we decided if that's to change?

**Comment by fcoveram** [comment]
> the search bar sticky, and is it only on this page?

No sticky search bar. The only sticky component in this page is the table of content.

I suggested [here](https://github.com/WordPress/wporg-mu-plugins/issues/465#issuecomment-1754693237) to only keep the local nav fixed, as in [the example](https://user-images.githubusercontent.com/541093/271435777-aa35957e-87d6-4e81-b0d2-15c8c12d9bd9.mp4) shared by @ryelle. But not sure whether that's being addressed or not.

**Comment by adamwoodnz** [comment]
> No sticky search bar. The only sticky component in this page is the table of content.

Ah ok. [This design](https://www.figma.com/file/2WxlJFzMJvqPfZL1EkAOVp/Developer-Resources?node-id=3735%3A19404&mode=dev) seems to contradict that, it shows header, chapter list and ToC sticky, unless I'm mistaken.

> I suggested [here](https://github.com/WordPress/wporg-mu-plugins/issues/465#issuecomment-1754693237) to only keep the local nav fixed, as in [the example](https://user-images.githubusercontent.com/541093/271435777-aa35957e-87d6-4e81-b0d2-15c8c12d9bd9.mp4) shared by @ryelle. But not sure whether that's being addressed or not.

Yeah I think maybe we need to get the header changes shipped before we can move forward with Developer and Documentation.

**Comment by jasmussen** [comment]
Ah, thanks for clarifying. Yes this was a mistake in the mockup showing "scrolled", I forgot the header. 

Here's what it would actually look like, with the sticky behavior as proposed, correct me if I'm wrong @fcoveram:

![Screenshot 2023-11-03 at 10 10 42](https://github.com/WordPress/wporg-developer/assets/1204802/fb4a8568-849c-479b-845e-5442a875e16e)

**Comment by fcoveram** [comment]
The version shown above is correct. Local nav fixed, and global header + breadcrumb non fixed.

**Comment by adamwoodnz** [comment]
Thanks for clarifying.

> Local nav fixed

Plus ToC, and chapters (looking at image above)?

Currently chapters are not, and we'd need to add the logic the ToC has for looking at page height and determining whether it can fit and be fixed, or needs to scroll because it's longer than the window. It'll by slightly more complex because it can be expanded.

**Comment by jasmussen** [comment]
Having the chapters and table of contents be sticky is an entirely not-urgent optimization we can do at any time, even after relaunch, and as with other similar elements would be the same across the site.


For what it's worth, Chapters could scroll independently. That is, be always 100% height and have `overflow-y: auto;`. It could be combined with a scrollbar like we use in the site editor.

<details><summary> Here's a mixin we use:</summary>

```
@mixin custom-scrollbars-on-hover($handle-color, $handle-color-hover) {

	// WebKit
	&::-webkit-scrollbar {
		width: 12px;
		height: 12px;
	}
	&::-webkit-scrollbar-track {
		background-color: transparent;
	}
	&::-webkit-scrollbar-thumb {
		background-color: $handle-color;
		border-radius: 8px;
		border: 3px solid transparent;
		background-clip: padding-box;
	}
	&:hover::-webkit-scrollbar-thumb, // This needs specificity.
	&:focus::-webkit-scrollbar-thumb,
	&:focus-within::-webkit-scrollbar-thumb {
		background-color: $handle-color-hover;
	}

	// Firefox 109+ and Chrome 111+
	scrollbar-width: thin;
	scrollbar-gutter: stable both-edges;
	scrollbar-color: $handle-color transparent; // Syntax, "dark", "light", or "#handle-color #track-color"

	&:hover,
	&:focus,
	&:focus-within {
		scrollbar-color: $handle-color-hover transparent;
	}

	// Needed to fix a Safari rendering issue.
	will-change: transform;

	// Always show scrollbar on Mobile devices.
	@media (hover: none) {
		& {
			scrollbar-color: $handle-color-hover transparent;
		}
	}
}
```

[It should look kinda like this](https://codepen.io/joen/pen/gOBLqJJ).

</details>

Let me know if you'd prefer I open a separate issue for this.

**Comment by adamwoodnz** [comment]
Thanks, that helps. I think we can get the ToC updated, it's pretty close.

> Let me know if you'd prefer I open a separate issue for this.

A new issue covering the chapter list sticky behaviour would be great please! We already have issues for collapsing it on mobile, and the show/hide behaviour.
